compiler-plugin/src/test/kotlin/arrow/meta/plugins/typeclasses/TypeClassesTest.kt

```diff
@@ -0,0 +1,83 @@
+package arrow.meta.plugins.typeclasses
+
+import arrow.meta.plugin.testing.CompilationTest
+import org.junit.Test
+
+class TypeClassesTest : CompilationTest {
+
+  @Test
+  fun `simple_case`() {
+    """
+    | import arrow.Kind
+    | import arrow.given
+    | import arrow.core.Some
+    | import arrow.core.Option
+    | import arrow.extension
+    | import arrow.core.ForOption
+    | import arrow.core.fix
+    | import arrow.core.None
+    |
+    | //metadebug
+    |
+    | @extension
+    | object OptionMappable : Mappable<ForOption> {
+    |   override fun <A, B> Kind<ForOption, A>.map(f: (A) -> B): Kind<ForOption, B> =
```

Comment:
@raulraja do we currently support `Option<A>` here instead already with the type checker?

If not should we continue using `OptionOf` instead? I've found this to be annoying to fix and maintain throughout the Arrow codebase.